src/adv.py

```diff
@@ -77,70 +79,77 @@ def start_game():
+                if command == 'n':
+                    if player.current_room.n_to is not None:
+                        player.current_room = player.current_room.n_to
+                    else:
+                        direction_error()
+                elif command == 's':
+                    if player.current_room.s_to is not None:
+                        player.current_room = player.current_room.s_to
+                    else:
+                        direction_error()
+                elif command == 'e':
+                    if player.current_room.e_to is not None:
+                        player.current_room = player.current_room.e_to
+                    else:
+                        direction_error()
+                elif command == 'w':
+                    if player.current_room.w_to is not None:
+                        player.current_room = player.current_room.w_to
+                    else:
+                        direction_error()
+                elif command == 'i' or command == 'inventory':
+                    if len(player.items) != 0:
+                        print('Inventory:')
+                        for items in player.items:
+                            print(items)
+                    else:
+                        print('You do not have any items in your inventory.')
                 else:
-                    print("Item entered is not available.")
-
-        def set_new_location(direction_input):
-            if direction_input == 'n':
-                if player.current_room.n_to is not None:
-                    player.current_room = player.current_room.n_to
-                else:
-                    print_error()
-
-            elif direction_input == 's':
-                if player.current_room.s_to is not None:
-                    player.current_room = player.current_room.s_to
-                else:
-                    print_error()
-            elif direction_input == 'e':
-                if player.current_room.e_to is not None:
-                    player.current_room = player.current_room.e_to
-                else:
-                    print_error()
-            elif direction_input == 'w':
-                if player.current_room.w_to is not None:
-                    player.current_room = player.current_room.w_to
-                else:
-                    print_error()
+                    print('Please enter a valid character.')
+            elif len(player_input) == 2:
+                item_action = player_input[0]
+                item_name = player_input[1]
+                global item
+                if item_action == 'take' or item_action == 'get':
+                    try:
+                        if item[item_name] in player.current_room.items:
+                            player.items.append(item[item_name])
+                            print(item[item_name].on_take())
+                            player.current_room.items.remove(item[item_name])
+                    except KeyError:
```

Comment:
Excellent use of `try-except` here as well as specifying `KeyError`. Impressive for your 4th day of python!